data-access-layer/metering/MeteringClient.js

```diff
@@ -0,0 +1,70 @@
+// Ignored this as jshint currently does not support async await
+/* jshint ignore:start */
+'use strict';
+
+const _ = require('lodash');
+const config = require('../../common/config');
+const utils = require('../../common/utils');
+const HttpClient = utils.HttpClient;
+const logger = require('../../common/logger');
+const CONST = require('../../common/constants');
+
+class MeteringClient extends HttpClient {
+
+  constructor(options) {
+    super(_.defaultsDeep({
+      baseUrl: config.metering.metering_url,
+      headers: {
+        Accept: 'application/json',
+      },
+      followRedirect: false
+    }, options));
+    this.clientId = config.metering.client_id;
+    this.clientSecret = config.metering.client_secret;
+    this.tokenUrl = config.metering.token_url;
+    this.meteringUrl = config.metering.metering_url;
+  }
+
+  async getAuthToken() {
+    try {
+      let res = await this
```

Comment:
Minor comment: Use `const` instead of `let` since the reference is not being changed